### PR TITLE
bugfix: avoided generating Content-Type header when getting/setting o…

### PR DESCRIPTION
**Pull request by spacewander**
…ther response headers.

After this change, we only generate Content-Type header when it is
got/set.

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

**Comment by spacewander** [comment]
@thibaultcha 
Updated.

File: src/ngx_http_lua_subrequest.c [hunk 1]
```diff
@@ -1031,6 +1031,14 @@ ngx_http_lua_post_subrequest(ngx_http_request_t *r, void *data, ngx_int_t rc)
...
 
     /* copy subrequest response headers */
+    if (ctx->headers_set) {
+        rc = ngx_http_lua_set_content_type(r);
+        if (rc != NGX_OK) {
+            ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+                          "failed to set default content type: %i", rc);
+            return NGX_ERROR;
+        }
+    }
```
Comment (thibaultcha) [inline src/ngx_http_lua_subrequest.c]: Are we lacking a test for this change? Reverting it does not seem to yield any failure in the subrequest test suite.

**Comment by thibaultcha** [comment]
Mostly looking good to me except for that one question about the subrequest change lacking test and my other question about "running the ngx_lua test suite with resty.core" vs. "porting some of the ngx_lua tests to resty.core" (raied [here](https://github.com/openresty/lua-resty-core/pull/218/files#r240501691)).

File: src/ngx_http_lua_subrequest.c (hunk 1 above)
Comment (spacewander) [reply src/ngx_http_lua_subrequest.c]: @thibaultcha 
The relative test cases are TEST 25 and TEST 26 in `002-content.t`.

Comment (thibaultcha) [reply src/ngx_http_lua_subrequest.c]: @spacewander Ah, perfect, thank you!

**Comment by thibaultcha** [comment]
LGTM, but let's wait for the sister PR's failing CI to get fixed first.

**Comment by thibaultcha** [comment]
Merged, thank you!

**Pull request closed without merging**